[TASK] Improve documentation on headline anchors (#448)
TYPO3-Documentation/TYPO3CMS-Guide-HowToDocument: Documentation/Reference/ReStructuredText/Menus/HeadlinesAndSection.rst

1. Use underlining plus overlining with `===` for the first section title of a
    file. The first section title is the "document title" (**doctitle**) of that
    file and will appear in the menu. Every .rst file should have a title, each

>     title should have a :ref:`link-anchor`:

Reviewer: ```suggestion
    title should have a link anchor. In :ref:`link-anchor` we describe 
    our recommended best way how to build a link anchor in the official 
    TYPO3 documentation. Here you see a title and a link anchor in
    action:
```

Length of Underlines
--------------------

> The length of the underlines **should** at have the length of the text. It

Reviewer: ```suggestion
The length of the underlines **should** have the same length like the text. It
```

**should not** be longer or shorter.

Example 1: This Works
~~~~~~~~~~~~~~~~~~~~~

..  code-block:: rst

    ..  _example-1:

    =========
    Example 1
    =========

Example 2: These headlines work but do not abide to the coding guidelines
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

> 

Reviewer: ```suggestion

Here we see how we should not do it. The underlines are too long or too short.

```

..  code-block:: rst

    ..  _example-2:

    ==============
    Example 2

>     ==============

Author: ```suggestion
    ==============
    
    # Underline too long
```

    ..  _example-3:

    =======
    Example 3

>     =======

Author: ```suggestion
    =======
    
    # Underline too short
```
